symbolic-base/src/ZkFold/Base/Protocol/Plonkup/Internal.hs

```diff
@@ -24,11 +23,11 @@ import           ZkFold.Symbolic.Compiler.ArithmeticCircuit.Internal
     Additionally, we don't want this library to depend on Cardano libraries.
 -}
 
-data Plonkup p (i :: Natural) (n :: Natural) (l :: Natural) curve1 curve2 transcript = Plonkup {
+data Plonkup p i (n :: Natural) l curve1 curve2 transcript = Plonkup {
```

Comment:
I am wondering if it makes sense to remove `n` as a type parameter. We cannot know the number of constraints in advance, so it might cause more confusion than good (as experienced, for instance, by @diS3e).

Reply:
Well, it makes sense to use it in some of the datatypes in Plonk, but there it can be introduced using [`withDict`](https://hackage.haskell.org/package/constraints-0.14.2/docs/Data-Constraint.html#v:withDict) with the exact number of constraints in the circuit

Reply:
So yes, in most cases it can be removed